### DISCUSSION TITLE
Access api resource method's metadata inside micronaut HttpServerFilter

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#5578] [Q&A] [answered]
With  jaxrs/jersey you can you can inject ResourceInfo into the server request interceptor(ContainerRequestFilter). This allows users to write a server request interceptor which can be specific to a certain resource method or path. 

With HttpServerFilter it can bind to a path. But inside the filter there's no way to access metadata of the matching resource method.
**example, How to read an annotation defined in the resource method  inside a server filter?.** 

How can I do that? (via jaxrs or via micronaut way)

**Comment by jameskleeh** [comment, marked as answer]
The route match is an attribute on the request. `request.getAttribute(HttpAttributes.ROUTE_MATCH, AnnotationMetadata.class)` That will provide you with an `Optional<AnnotationMetadata>` which contains the annotations on the method if available.
